src/postgraphile/http/createPostGraphileHttpRequestHandler.ts

```diff
@@ -1,30 +1,49 @@
 /* eslint-disable */// Because we use tslint
+/* tslint:disable:no-any no-var-requires */
 import {
   join as joinPath,
   resolve as resolvePath,
   relative as relativePath,
 } from 'path'
 import { readFile } from 'fs'
-import { IncomingMessage, ServerResponse } from 'http'
 import {
   Source,
   parse as parseGraphql,
   validate as validateGraphql,
   execute as executeGraphql,
-  getOperationAST,
   formatError as defaultFormatError,
   print as printGraphql,
   specifiedRules,
+  GraphQLError,
 } from 'graphql'
+import { EventEmitter } from 'events'
 import { extendedFormatError } from '../extendedFormatError'
-import { $$pgClient } from '../../postgres/inventory/pgClientFromContext'
-import renderGraphiQL from './renderGraphiQL'
-import debugPgClient from './debugPgClient'
+import { GraphQLSchema } from 'graphql'
+import { IncomingMessage, ServerResponse } from 'http'
+import { pluginHookFromOptions } from '../pluginHook'
+import { Pool } from 'pg'
+import { Postgraphile } from '../../interfaces'
+import HttpRequestHandler = Postgraphile.HttpRequestHandler
 import setupServerSentEvents from './setupServerSentEvents'
-import setupPgClientTransaction from '../setupPgClientTransaction'
 import withPostGraphileContext from '../withPostGraphileContext'
-import mapKeys from 'lodash/mapKeys'
-import { pluginHookFromOptions } from '../pluginHook'
+
+/**
+ * Creates a GraphQL request handler that can support many different `http` frameworks, including:
+ *
+ * - Native Node.js `http`.
+ * - `connect`.
+ * - `express`.
+ * - `koa` (2.0).
+ */
+export interface ICreateRequestHandler extends Postgraphile.PostGraphileOptions {
```

Comment:
boom! that huge red-file (`createPostGraphileHttpRequestHandler.d.ts`), gets compacted down to these 9 LOC. mo betta, much betta.